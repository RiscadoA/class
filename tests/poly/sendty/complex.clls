proc main() {
    cut {
        println("sendty"); sendty x(close);
        println("lhs send"); send x(y. println("close y"); close y);
        println("lhs recv"); recv x(y);
        println("lhs wait"); wait x;
        println("wait y"); wait y; ()
        |x: recvty T; recv ~T; send T; close|
        println("recvty"); recvty x(U);
        println("rhs recv"); recv x(y);
        cut {
            println("forward");
            fwd y z
            |z: ~U|
            println("rhs send"); send x(z);
            println("rhs close"); close x
        }
    }
};;
