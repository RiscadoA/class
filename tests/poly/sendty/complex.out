sendty
lhs send
close y
recvty
rhs recv
forward
rhs send
rhs close
lhs recv
lhs wait
wait y
